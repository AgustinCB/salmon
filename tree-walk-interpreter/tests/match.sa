fun withmatch(v) {
  match v {
    1 => {
      print "uno";
    },
    2 => {
      print "dos";
    },
    nil => {
      print "nulo";
    },
    true => {
      print "verdadero";
    },
    * => {
      print v;
    },
  }
}

withmatch(1);
withmatch(2);
withmatch(nil);
withmatch(true);
withmatch("hola");
