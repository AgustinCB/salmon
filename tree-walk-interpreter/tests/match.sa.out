uno
dos
nulo
verdadero
hola
